tests/test_project/test_app/tests.py

```diff
@@ -219,3 +223,44 @@ def test_refresh_materialised_doesnt_error(self):
         # and it could lead to a flakey test. We instead check the function doesn't error.
         refresh_materialized_view(SimpleMaterializedView(), concurrently=True)
 
+
+class TestReadableViews(BaseTestCase):
+
+    @staticmethod
+    def _add_row(**kwargs):
+        default_fields = {
+            "integer_col": 1,
+            "character_col": "a",
+            "date_col": datetime.date(2019, 1, 1),
+            "datetime_col": datetime.datetime(2019, 1, 1)
+        }
+        return TestModel.objects.create(**default_fields, **kwargs)
+
+    def test_readable_view_from_queryset_no_results(self):
+        results = ReadableTestViewFromQueryset.objects.values('id')
+        self.assertEqual(results.count(), 0)
+
+    def test_readable_view_from_queryset_with_results(self):
+        test_data = self._add_row()
+        results = list(ReadableTestViewFromQueryset.objects.values_list('id', flat=True))
+        self.assertEqual(results, [test_data.id])
+
+    def test_readable_view__does_not_exist_exception_raised_when_object_not_found(self):
+        with self.assertRaises(ReadableTestViewFromQueryset.DoesNotExist):
+            ReadableTestViewFromQueryset.objects.get(id=123)
+
+    def test_readable_view__multiple_objects_exception_raised_when_multiple_objects_found(self):
+        test_data_1 = self._add_row()
+        test_data_2 = self._add_row()
+        with self.assertRaises(ReadableTestViewFromQueryset.MultipleObjectsReturned):
+            ReadableTestViewFromQueryset.objects.get(character_col=test_data_1.character_col)
+
+    def test_readable_view_from_sql_no_results(self):
+        results = ReadableTestViewFromSQL.objects.values('id')
+        self.assertEqual(results.count(), 0)
+
+    def test_readable_view_from_sql_with_results(self):
+        test_data = self._add_row()
+        results = list(ReadableTestViewFromSQL.objects.values_list('id', flat=True))
+        self.assertEqual(results, [test_data.id])
+
```

Comment:
Could we test with a view with a foreign key and a lookup on that?  i.e. make a view that FKs to a model

Reply:
Done